packages/static/kairos-overlay-files/files/system/oem/00_datasource.yaml

```diff
@@ -2,7 +2,7 @@ name: "Datasource handling"
 stages:
   rootfs.before:
     - &datasource
-      if: '[ ! -f /oem/userdata ] && [ ! -f /run/cos/uki_boot_mode ]'
+      if: '[ ! -f /oem/userdata ] && ([ ! -f /run/cos/uki_boot_mode ] || grep -q "kairos.pull_datasources" /proc/cmdline )'
```

Comment:
It doesn't seem to work:

```
root@localhost:/home/kairos# cat /oem/userdata
#cloud-config

stages:
  boot:
    - name: "Create a file from datasource"
      commands:
        - touch /etc/datasource-file-dimitris
root@localhost:/home/kairos# ls /etc/datasource-file-dimitris
/etc/datasource-file-dimitris
root@localhost:/home/kairos# cat /proc/cmdline
console=ttyS0 console=tty1 net.ifnames=1 rd.immucore.oemlabel=COS_OEM rd.immucore.oemtimeout=2 rd.immucore.uki selinux=0 panic=5 rd.shell=0 systemd.crash_reboot=yes install-mode
root@localhost:/home/kairos# ls  /run/cos/uki_boot_mode
/run/cos/uki_boot_mode
```

No idea why.

Reply:
```
root@localhost:/home/kairos# if [ ! -f /run/cos/uki_boot_mode ] || grep -q "kairos.pull_datasources" /proc/cmdline; then echo 'ok'; fi
root@localhost:/home/kairos#
```
manually the conditions seems to work :shrug:

Reply:
It was my mistake, thinking the `/oem/userdata` file is re-created on every boot. I initially booted the entry that had the cmdline option and then rebooted to the entry that didn't. But the file was already there :facepalm: . If I delete the file and boot the entry with no option, the file is not created. All good.